### ISSUE TITLE
Multipart file corruption recovery

### ISSUE THREAD

**Issue by osmano807** [fcorbelli/zpaqfranz#28]
Say I'm using multipart files and inadvertently delete a version. 
What's the supposed route to recover? Operations on the file gives errors:
```
❯ zpaqfranz l 'mainbackup_????.zpaq' -index mainbackup_0000.zpaq -filelist
zpaqfranz v55.10f archiver,  compiled Jan  1 1980
franz:-flagfilelist
11896:mainbackup_0001.zpaq: No such file or directory
23013: zpaqfranz error: mainbackup_0001.zpaq

0.039 seconds (00:00:00)  (with errors)
```
How should I proceed to remove the dependency on that version?

When substituting the missing file with the index, I can list but not extract files:
```
❯ zpaqfranz l 'mainbackup_????.zpaq' -index mainbackup_0000.zpaq -filelist
zpaqfranz v55.10f-experimental-5078cf5dcd0b04c7f362a9ee406d077c77732d02 archiver,  compiled Jan  1 1980
franz:-flagfilelist
mainbackup_????.zpaq:
Unordered fragment tables: expected >= 7 found 3
Unordered fragment tables: expected >= 7 found 5
5 versions, 15 files, 6 fragments, 15 blocks, 5.168 bytes (5.05 KB)

- 2022-09-04 15:46:26                 217  0644 VFILE-l-filelist.txt
- 2022-09-04 15:46:25                   0 d0755 bkp/
- 2022-09-04 15:45:22                   3  0644 bkp/01
- 2022-09-04 15:46:18                   3  0644 bkp/02
- 2022-09-04 15:46:25                   3  0644 bkp/03

                  226 (226.00 B) of 226 (226.00 B) in 5 files shown
                6.695 compressed
Note: 1.027 of 2.554 compressed bytes are in archive

0.001 seconds (00:00:00)  (all OK)
```
```
❯ zpaqfranz x 'mainbackup_????.zpaq' -to extraction/
zpaqfranz v55.10f-experimental-5078cf5dcd0b04c7f362a9ee406d077c77732d02 archiver,  compiled Jan  1 1980
mainbackup_????.zpaq:
Unordered fragment tables: expected >= 7 found 3
Unordered fragment tables: expected >= 7 found 5
5 versions, 15 files, 6 fragments, 15 blocks, 5.168 bytes (5.05 KB)
23013: zpaqfranz error: unordered frags

0.001 seconds (00:00:00)  (with errors)

❯ ls extraction/

❯ 
```

Also, how do I recreate an index file for such use?

**Comment by fcorbelli** [comment]
Multipart archives are exactly a splitted file  
Example
arc_01.zpaq 10.000 bytes  
arc_02.zpaq 3.000 bytes  
arc_03.zpaq 2.000 bytes  

If you "consolidate" with "cat" or "copy /b" or whatever (or using the m command (merge)) you get a
total.zpaq=15.000 bytes  

Therefore if part X is missing, everything BEFORE X (upto X-1) still works just fine.  
From >=X it really depends on "what was inside" piece X and X+1  

With lost D-blocks (data fragments) any files with fragment inside "here" are lost
The same thing for H (list of fragment hashes, pointing D blocks)
For I (index) say "the directory" every file is lost (cannot retrieve I and D)

It essentially depends on deduplication, and therefore on whether or not fragments are added AND from file information (I)  
If in version X-1 you do not have pippo.txt, and you write in version X, then you will never be able to restore, because you will not "know" that a pippo.txt exists (in this example you DO NOT have the data, too)

In general, as a rule, all> X is lost ("unordered frags" is a "kaputt")  

**Short version: I do not think you can recover in this case**  

You can try asking Matt Mahoney (the original developer) what he recommends (maybe he may have better ideas than mine). He is a very courteous person, sometimes he answers
